x-pack/plugins/observability/public/data_handler.ts

```diff
@@ -31,26 +32,26 @@ export function getDataHandler<T extends ObservabilityFetchDataPlugins>(appName:
   }
 }
 
-export async function fetchHasData(): Promise<Record<ObservabilityFetchDataPlugins, boolean>> {
-  const apps: ObservabilityFetchDataPlugins[] = ['apm', 'uptime', 'infra_logs', 'infra_metrics'];
-
-  const promises = apps.map(async (app) => getDataHandler(app)?.hasData() || false);
+export function useInfraLogsHasData() {
+  return useFetcher(() => getDataHandler('infra_logs')?.hasData(), []);
+}
 
-  const results = await Promise.allSettled(promises);
+export function useInfraMetricsHasData() {
+  return useFetcher(() => getDataHandler('infra_metrics')?.hasData(), []);
+}
 
-  const [apm, uptime, logs, metrics] = results.map((result) => {
-    if (result.status === 'fulfilled') {
-      return result.value;
-    }
+export function useApmHasData() {
+  return useFetcher(() => getDataHandler('apm')?.hasData(), []);
+}
 
-    console.error('Error while fetching has data', result.reason);
-    return false;
-  });
+export function useUptimeHasData() {
+  return useFetcher(() => getDataHandler('uptime')?.hasData(), []);
+}
 
-  return {
-    apm,
-    uptime,
-    infra_logs: logs,
-    infra_metrics: metrics,
-  };
+export function useUxHasData({ start, end }: { start: number; end: number }) {
+  return useFetcher(
+    () => getDataHandler('ux')?.hasData({ absoluteTime: { start, end } }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  );
```

Comment:
I think there's a mix of responsibilities here. This file now hosts code to register/unregister data handler and react hooks. Maybe you could create a `useDataHandler` hook and place this code there?